### PR TITLE
Feature/sm science improvements

### PR DESCRIPTION
**Pull request by smahanam**
This PR is related to the updates in [PR 151](https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/151)

### CODE REVIEW

File: src/Applications/LDAS_App/GEOSldas_LDAS.rc [hunk 1]
```diff
@@ -152,3 +146,14 @@ LAND_ASSIM                     : NO
...
+# ---- Surface layer turbulence scheme:
+#    0 : Louis                  (MERRA, Fortuna-DAS, SMAP NRv4/4.1/5/7.2)
+#    1 : Helfand Monin-Obukhov  (Fortuna-AR5, Ganymed, Heracles, Icarus-3_2, MERRA-2)
+#
+CHOOSEMOSFC                    : 0
+
+# Fraction of AGCM grid cell that precipitation falls on
+# ------------------------------------------------------
+# options : 1.0 (default for zero-diff); 0.025 is recommended to deal with AGCM's drizzle
+# PRECIPFRAC: 0.025
```
Comment (gmao-rreichle) [inline src/Applications/LDAS_App/GEOSldas_LDAS.rc]: Github complains that there's no newline character at the end of the file.  To be safe, it's probably a good idea to add a newline character.

**Comment by gmao-rreichle** [comment]
Is this GEOSldas PR required if GCM PR#151 is merged in order to keep GEOSldas working?

Also, why would these changes need to be merged on "master"?  Are they already on "develop"?
Or am I misreading the PR?

**Comment by smahanam** [comment]
My apologies .... I am closing, this PR should go to develop

**Comment by smahanam** [comment]
To your question "Is this GEOSldas PR required if GCM PR#151 is merged in order to keep GEOSldas working?" @gmao-rreichle 
Yes there are few things from top of my head
1) peat-CLM needs bf1, bf2 to compute soil moisture (so there are few calls from GEOSldas to calc_soil_moist (2 new arguments)
2) CATCHCN scale para implementation needs few new BCs files, and I don't see the point in creating those files for each BCs dir now. So I have put a condition to check availability of those files in BCs dir.
3) CATCHCN PRESCRIBE_DVG - if GEOSldas wants to run, it sure needs these modifications in lenkf.j.template
4) I took out a bunch of parameters from GEOSldas_LDAS.rc and moved to new GEOS_LandGridComp.rc 
We GEOSldas readers use the sample text file to change tunable parameters. This modified ldas_setup reads and dump both  GEOSldas_LDAS.rc and GEOS_LandGridComp.rc to write the sample file
5) Also, MAPL_GetResource was replaced with ESMF_GetConfigAttribute calls in the model.
Here is the trick. 
AGCM.rc has a line 
LANDRC: GEOS_LandGridComp.rc
while LDAS.rc has a line 
LANDRC: LDAS.rc (itself) 
Basically, for GEOSldas user's, they don't have to worry about anything this new GEOS_LandGridComp.rc. AGCM users, like they usually do for other cases, they will need to edit RC/GEOS_LandGridComp.rc after the experiment was created. 

So the simple answer, yes these modifications are needed. I will close the PR and send a PR to develop.
